### PR TITLE
BLD: initial build of stylegan generator

### PR DESCRIPTION
**Pull request by Jped**
New Roll using style gan structure.

### CODE REVIEW

**Comment by eigenfoo** [comment]
You should merge master into this branch, so that you get the latest changes. This should do the trick:

```
git checkout master
git pull origin master
git checkout style_roll
git merge master
```

You'll have to work through any merge conflicts.

**Comment by eigenfoo** [comment]
#76 should do this for you: merge when you've taken a look at it.

File: src/matzah.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        super(StyleNetwork, self).__init__()
+        self.fc1 = nn.linear(styleDim, fcSize)
+        self.fcLayers = [
+            nn.linear(fcSize, fcSize) for i in range(1, len(fcSizes))
+        ]
+        self.relu = nn.ReLU()
+
+    def forward(self, styleVec, classVec):
+        # concate the classVec and the StyleVec
+        style = torch.cat((styleVec, classVec), dim=0)
```
Comment (eigenfoo) [inline src/matzah.py]: This isn't right. We should concatenate the one-hot class vector to the output of the dense layers.

```python
x = nn.Linear(styleVec)
x = nn.Linear(x)
x = nn.Linear(x)
return torch.cat([x, classVec], dim=0)
```

Comment (Jped) [reply src/matzah.py]: This is an easy fix, but what would be the practical difference? Would there be an issue with doing it this way.

I'm changing it right now, but just want to have the discussion

**Pull request merged**